src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -1,14 +1,23 @@
 package wolox.training.repositories;
 
+import java.time.LocalDate;
+import java.util.List;
 import java.util.Optional;
 
 import javax.validation.constraints.NotBlank;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 import wolox.training.models.User;
 
 public interface UserRepository extends JpaRepository<User, Long> {
 
 	Optional<User> findFirstByUserName(@NotBlank String userName);
+
+	@Query("select u from User u where u.birthDate between :dateFrom and :dateTo and u.name like :cadena")
+	List<User> findByBirthDateBetweenAndCadena(@Param("dateFrom") LocalDate dateFrom, @Param("dateTo") LocalDate dateTo,
+	        @Param("cadena") String cadena);
```

Comment:
Tres comentarios en este punto :)
1. Usemos valores en ingles para las variables, en lugar de cadena podría ser "name"
2. El nombre del método habría que cambiarlo para que siga el cambio de variable de "cadena" por "name" también
3. Qué pasa si nos mandan fechas en null en este caso? O nos mandan sólo una? Creo que no contemplamos ese caso. Para chequear si una fecha es null tendríamos que castearla y parte de la condición quedaría así: ...`OR (cast(:startDate as date) IS NULL`